### ISSUE TITLE
A few accessibility bugs that I've noticed when experimenting (as well as some suggestions)

### ISSUE THREAD

**Issue by ethindp** [godotengine/godot#117197]
### Tested versions

Godot Engine v4.6.1.stable.official.14d19694e

### System information

Godot v4.6.1.stable - Windows 10 (build 19044) - Multi-window, 1 monitor - Direct3D 12 (Forward+) - dedicated Intel(R) Iris(R) Xe Graphics (Intel Corporation; 27.20.100.9664) - 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz (8 threads) - 39.84 GiB memory

### Issue description

This issue is my notes of a few accessibility problems in Godot which require fixing. I wasn't sure how exactly to file this, so this also contains some suggestions. It is worth noting that my screen reader of choice is NVDA, so I don't know what the experience is like with others.

1. When opening sub-inspectors (e.g., when editing an audio stream which is sNOT an audio file), the focus is yanked to an anonymous, empty, focus-capturing editor node. The only way to get out of this is by using object nav in NVDA. The focus needs to be forced back to the edit button where the user was originally at or, failing that, to the sub-inspectors first focusable control.
2. Autocomplete in the script editor is not accessible. Idk if this is possible, but could we make it like the search field in the windows Settings app or the start menu? NVDA can naturally detect these pop-up dialogs, as well as when they open and close, and that makes accessibility much nicer.
3. Certain dialogs, such as the rename node and color select dialog, are not accessible. The color select dialog is the one displayed when you want to select the color property of things like labels and their modulate property.
4. AccessKit should be updated to it's latest version. This will probably bring about some magical accessibility improvements.
5. The UI is currently quite cluttered from the POV of a screen reader. Some thoughts on improving this:
    * A shortcut could be added (e.g., control comma and period) to navigate to the previous and next regions of the editor. A region would be, e.g., the inspector, the node tree, the file system doc, etc.
    * The editor could be expanded to allow you to customize what it contains (this is somewhat already doable)
6. The menu bar is not focusable with the alt key as would be doable in a standard app. Additionally, accelerator keys don't work as would be expected. This makes using things that depend on it rather annoying.

### Steps to reproduce

Build even a somewhat complicated project with Godot. Navigating between the inspector and scene/node tree is tedious because there are so many widgets in the way. For the sub-inspector issue, create an `AudioStreamPlayer3D` node and set the Stream property to `AudioStreamPolyphonic`. Then hit Edit and Edit. You'll be yanked into some node that I don't think actually exists.

For the color dialog, you can just try modifying the RGBA color of anything. Same for the rename dialog.

### Minimal reproduction project (MRP)

N/A
